src/plugins/data_source/server/routes/fetch_data_source_version.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter, OpenSearchClient } from 'opensearch-dashboards/server';
+import { AuthType, DataSourceAttributes, SigV4ServiceName } from '../../common/data_sources';
+import { DataSourceConnectionValidator } from './data_source_connection_validator';
+import { DataSourceServiceSetup } from '../data_source_service';
+import { CryptographyServiceSetup } from '../cryptography_service';
+import { IAuthenticationMethodRegistery } from '../auth_registry';
+import { CustomApiSchemaRegistry } from '../schema_registry/custom_api_schema_registry';
+
+export const registerFetchDataSourceVersionRoute = async (
+  router: IRouter,
+  dataSourceServiceSetup: DataSourceServiceSetup,
+  cryptography: CryptographyServiceSetup,
+  authRegistryPromise: Promise<IAuthenticationMethodRegistery>,
+  customApiSchemaRegistryPromise: Promise<CustomApiSchemaRegistry>
+) => {
+  const authRegistry = await authRegistryPromise;
+  router.post(
+    {
+      path: '/internal/data-source-management/fetchDataSourceVersion',
+      validate: {
+        body: schema.object({
+          id: schema.maybe(schema.string()),
+          dataSourceAttr: schema.object({
+            endpoint: schema.string(),
+            auth: schema.maybe(
+              schema.oneOf([
+                schema.object({
+                  type: schema.literal(AuthType.NoAuth),
+                  credentials: schema.object({}),
+                }),
+                schema.object({
+                  type: schema.literal(AuthType.UsernamePasswordType),
+                  credentials: schema.object({
+                    username: schema.string(),
+                    password: schema.string(),
+                  }),
+                }),
+                schema.object({
+                  type: schema.literal(AuthType.SigV4),
+                  credentials: schema.object({
+                    region: schema.string(),
+                    accessKey: schema.string(),
+                    secretKey: schema.string(),
+                    service: schema.oneOf([
+                      schema.literal(SigV4ServiceName.OpenSearch),
+                      schema.literal(SigV4ServiceName.OpenSearchServerless),
+                    ]),
+                  }),
+                }),
+              ])
+            ),
+          }),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const { dataSourceAttr, id: dataSourceId } = request.body;
```

Comment:
since data source id is optional, how do we handle undefined data source id here?

Reply:
same as the test connection call , dataSourceId is actually needed for the OS API call but more on the MD internal side for validation to make sure the call is actually triggered by data source creation.